toucan_connectors/toucan_connector.py

```diff
@@ -36,3 +36,16 @@ def __init_subclass__(cls):
     @abstractmethod
     def get_df(self, data_source: ToucanDataSource) -> pd.DataFrame:
         """Main method to retrieve a pandas dataframe"""
+
+    def get_df_and_count(self, data_source: ToucanDataSource, limit: int):
```

Comment:
The mongo connector implementation suggested that `limit` could be None. Is this actually the case? If so, the type annotation should probably be updated.